File: Source/Easy_Closest_Point.c
#include <math.h>
#include <stdio.h>

struct Point
{
    int x;
    int y;
};

double Distance(struct Point point)
{
    return sqrt(pow(fabs(point.x), 2) + pow(fabs(point.y), 2));
}

struct Point Closest_Point(struct Point* points, unsigned int size)
{
    struct Point closest_point = {0,0};
    if(size == 0) return closest_point;
    closest_point = points[0];

    for(unsigned int i = 0; i < size; i++)
    {
        if(Distance(points[i]) < Distance(closest_point)) closest_point = points[i];
    }

    return closest_point;
}

struct Point points[] = {{2,3},{4,9},{10,10},{1,1}};
unsigned int size = sizeof(points) / sizeof(struct Point);

int main()
{
    struct Point point = Closest_Point(points, size);
    printf("Closest Point: [%d,%d]\n", point.x, point.y);
}

File: Source/Easy_Closest_Point.cs
using System;
using System.Collections.Generic;

class Point
{
    public int x = 0;
    public int y = 0;

    public Point(int x, int y)
    {
        this.x = x;
        this.y = y;
    }
}

class Calculate
{
    public static double Distance(Point point)
    {
        return Math.Sqrt(Math.Pow(Math.Abs(point.x), 2) + Math.Pow(Math.Abs(point.y), 2));
    }
}

class Algorithm
{
    public static Point Closest_Point(List<Point> points)
    {
        if(points.Count == 0) return new Point(0,0);
        Point closest_point = new Point(0,0);
        closest_point = points[0];

        for(int i = 0; i < points.Count; i++)
        {
            if(Calculate.Distance(points[i]) < Calculate.Distance(closest_point))
            {
                closest_point = points[i];
            }
        }

        return closest_point;
    }
}

class Program
{
    static void Main()
    {
        List<Point> points = new List<Point>();
        points.Add(new Point(8,8));
        points.Add(new Point(4,4));
        points.Add(new Point(2,2));
        points.Add(new Point(1,1));
        Point point = Algorithm.Closest_Point(points);
        Console.WriteLine("Closest Point: [{0},{1}]", point.x, point.y);
    }
}